code/game/machinery/tcomms/_base.dm

```diff
@@ -0,0 +1,412 @@
+/*
+
+	ParadiseSS13 Telecommunications System
+
+	Rewritten from the ground up because I was totally unhappy with how laggy and complicated the current implementation was
+	The system is made up of two objects. A main core and relays.
+
+	The main core is basically the same as all of the machines from the previous implementation, apart from the relay
+	The core handles recieving and sending messages, logging messages, the NTTC configuration, and serves as the linkage hub for relays
+
+	Relays function much in the same way as the old ones. They just expand the reach of tcomms from one z-level to another.
+
+	This file contains extra datums and helper procs which the system utilises.
+	Unlike old telecomms, everything in here **should** be well documented. If not, feel free to add your own
+	-aa07
+
+*/
+
+/// Global list for all telecomms machines in the world
+GLOBAL_LIST_EMPTY(tcomms_machines)
+
+/**
+  * # Telecommunications Device
+  *
+  * This is the base machine for both tcomms devices (core + relay)
+  *
+  * This holds a few base procs (Icon updates, enable/disable, etc)
+  * It also has the initial overrides for Initialize() and Destroy()
+  */
```

Comment:
For those wondering why I have done this style of comment and triple-slashes in places, this is the syntax that the autodoc generator uses.